AFQ/utils/volume.py

```diff
@@ -42,3 +49,62 @@ def patch_up_roi(roi, bundle_name="ROI"):
         return convex_hull_image(hole_filled)
     except QhullError:
         return hole_filled
+
+
+def density_map(tractogram, n_sls=None, to_vox=False):
+    """
+    Create a streamline density map.
+    based on:
+    https://dipy.org/documentation/1.1.1./examples_built/streamline_formats/
+
+    Parameters
+    ----------
+    tractogram : StatefulTractogram
+        Stateful tractogram whose streamlines are used to make
+        the density map.
+    n_sls : int or None
+        n_sls to randomly select to make the density map.
+        If None, all streamlines are used.
+        Default: None
+    to_vox : bool
+        Whether to put the stateful tractogram in VOX space before making
+        the density map.
+
+    Returns
+    -------
+    Nifti1Image containing the density map.
+    """
+    if to_vox:
+        tractogram.to_vox()
+
+    sls = tractogram.streamlines
+    if n_sls is not None:
+        sls = select_random_set_of_streamlines(sls, n_sls)
+
+    affine, vol_dims, voxel_sizes, voxel_order = get_reference_info(tractogram)
+    tractogram_density = dtu.density_map(sls, np.eye(4), vol_dims)
+    nifti_header = create_nifti_header(affine, vol_dims, voxel_sizes)
+    density_map_img = nib.Nifti1Image(tractogram_density, affine, nifti_header)
+
+    return density_map_img
+
+
+def dice_coeff(img1, img2):
+    """
+    Compute Dice's coefficient between two imagess.
+
+    Parameters
+    ----------
+    img1 : Nifti1Image
+        One image to compare.
+    img2 : Nifti1Image
+        The other image to compare.
+
+    Returns
+    -------
+    The dice similarity between the images.
+    """
+    # scipy's dice function returns the dice *dissimilarity*
+    return 1 - dice(
+        img1.get_fdata().flatten().astype(bool),
```

Comment:
This assumes that the input is a nibabel image, but I think that it would be nice to allow passing full path, nibabel image or array. You can check which you got on the input and then proceed as necessary: if it's an array, you go ahead and  calculate Dice. If it's a nibabel image, you need to call `get_fdata` and then calculate Dice. If it's a string, you read the  file, call  `get_fdata` and then calculate Dice.